### PR TITLE
[MOB 551] - Back button

### PR DESCRIPTION
**Pull request by devinroche**
Add back button to setup wallet page.

### CODE REVIEW

File: src/components/Auth/AuthNavigator.js [hunk 1]
```diff
@@ -5,8 +5,8 @@ import wrapNavigator, { routeKeys } from 'components/shared/wrapNavigator';
 import Loading from './Loading';
 import Mnemonic from './Mnemonic';
 import PinSetup from './PinSetup';
-import Ready from './Ready';
```
Comment (lysyi3m) [inline src/components/Auth/AuthNavigator.js]: Why do we need to drop `Auth/Ready` route here? This screen is important step of signup process: https://github.com/mobius-network/wallet/blob/079e259ce9336956124e7c79b0c1824962d6864e/src/state/sagas/auth/recoveryFinishSaga.js#L31

File: src/components/Auth/AuthNavigator.js [hunk 2]
```diff
@@ -5,8 +5,8 @@ import wrapNavigator, { routeKeys } from 'components/shared/wrapNavigator';
...
 import Recovery from './Recovery';
+import Choose from './PinSetup/PinPad/Choose';
```
Comment (lysyi3m) [inline src/components/Auth/AuthNavigator.js]: What `Auth/Choose` route stands for? I can not see any actual routing to that screen

File: src/components/Auth/PinSetup/PinPad/Choose/Setup/Setup.js [hunk 3]
```diff
@@ -28,15 +32,19 @@ class Setup extends Component {
...
     const { pinLength, subtitle, title } = this.props;
     const { pin } = this.state;
 
     return (
       <Fragment>
         <Header
+          handleBack={this.handleBack}
           pin={pin}
           pinLength={pinLength}
+          showBack={true}
```
Comment (lysyi3m) [inline src/components/Auth/PinSetup/PinPad/Choose/Setup/Setup.js]: Could be written like `<Header showBack />`

File: src/components/Auth/AuthNavigator.js (hunk 1 above)
Comment (devinroche) [reply src/components/Auth/AuthNavigator.js]: Hmm I didn't mean to do this! Ill get that fixed, my bad!

File: src/components/Auth/AuthNavigator.js [hunk 4]
```diff
@@ -5,8 +5,8 @@ import wrapNavigator, { routeKeys } from 'components/shared/wrapNavigator';
 import Loading from './Loading';
 import Mnemonic from './Mnemonic';
 import PinSetup from './PinSetup';
-import Ready from './Ready';
 import Recovery from './Recovery';
+import Ready from './Ready';
```
Comment (lysyi3m) [inline src/components/Auth/AuthNavigator.js]: Let's do not commit unnecessary changes :)

File: src/components/Auth/PinSetup/PinPad/Header/styles.js [hunk 5]
```diff
@@ -1,4 +1,6 @@
 import { StyleSheet } from 'react-native';
+import styled from 'styled-components';
```
Comment (lysyi3m) [inline src/components/Auth/PinSetup/PinPad/Header/styles.js]: It's okay to use `styled-components` here, but it's not a good idea to mix `StyleSheet.create()` with `styled-components` together in the same file. It would be better to refactor a bit existing styles using `styled-components` first - just because of consistency and maintainability

Comment (devinroche) [reply src/components/Auth/PinSetup/PinPad/Header/styles.js]: Sounds good I'll move the styles to styled components.

**Comment by devinroche** [comment]
Hi @lysyi3m not sure if you get updates if I push changes, but I just updated the branch, feel free to add more suggestions! 😸

File: src/components/Auth/PinSetup/PinPad/Header/styles.js [hunk 6]
```diff
@@ -9,89 +8,73 @@ import {
...
+
+export const Subtitle = styled.Text`
+  font-size: ${fontSizes.default};
+  line-height: ${helpers.calculateLineHeight(fontSizes.default)};
+  padding-top: 10;
+  position: absolute;
+  text-align: center;
+  top: 100%;
+  width: ${helpers.vw - 48};
+  color: ${props => (props.showError ? colors.error : colors.textSecondary)};
```
Comment (lysyi3m) [inline src/components/Auth/PinSetup/PinPad/Header/styles.js]: This will cause "unknown prop `showError` on <Text>"  like warnings. We'll need to workaround this using props filtering like this https://github.com/styled-components/styled-components/issues/135

File: src/components/Auth/PinSetup/PinPad/Header/styles.js [hunk 7]
```diff
@@ -117,3 +100,13 @@ export const BackIcon = styled(MaterialIcon).attrs({
...
   underlayColor: 'transparent',
 })``;
+
+export const StyledDot = styled.View`
+  border-radius: 10;
+  height: 20;
+  margin-left: 9;
+  margin-right: 9;
+  width: 20;
+  border-color: ${props => (props.showError ? colors.error : '#BEBEBE')};
```
Comment (lysyi3m) [inline src/components/Auth/PinSetup/PinPad/Header/styles.js]: Same here

File: src/components/Auth/PinSetup/PinPad/Header/styles.js [hunk 8]
```diff
@@ -7,83 +8,105 @@ import {
...
+  width: ${helpers.vw - 48};
+`;
+
+export const Title = styled.Text`
+  flex: 1;
+  flex-wrap: wrap;
+  font-family: ${fonts.helveticaNeue.bold};
+  font-size: ${fontSizes.secondaryTitle};
+  text-align: center;
+  color: ${props => (props.showError ? colors.error : colors.textDefault)};
```
Comment (devinroche) [inline src/components/Auth/PinSetup/PinPad/Header/styles.js]: @lysyi3m Still trying to figure out the "unknown props" warning, but im assuming this would also show an error?

**Comment by devinroche** [comment]
@lysyi3m Sweet I got the solution (or I think so, I don't see the warning anymore.)
